marlowe-chain-sync/src/Language/Marlowe/Runtime/Cardano/Api.hs

```diff
@@ -276,6 +276,16 @@ toCardanoTxOutDatum era = curry case featureInCardanoEra era of
     (Just hash, Nothing) -> C.TxOutDatumHash scriptDataSupported <$> toCardanoDatumHash hash
     (_, Just datum) -> Just $ C.TxOutDatumInTx scriptDataSupported $ toCardanoScriptData datum
 
+toCardanoTxOutDatum'
+  :: C.CardanoEra era
+  -> Maybe DatumHash
+  -> Maybe (C.TxOutDatum ctx era)
+toCardanoTxOutDatum' era = case featureInCardanoEra era of
+  Nothing -> const $ Just C.TxOutDatumNone
+  Just scriptDataSupported -> \case
+    Nothing -> Just C.TxOutDatumNone
+    Just hash -> C.TxOutDatumHash scriptDataSupported <$> toCardanoDatumHash hash
```

Comment:
This is a workaround for a design flaw in `Cardano.Api` where forces `TxOutDatumInTx` to `CtxTx` instead of `ctx`, but it hides `TxOutDatumInTx'`.